### ISSUE TITLE
cppcheck broken when used with some addons (misra for example)

### ISSUE THREAD

**Issue by alfred2g** [dense-analysis/ale#3351]
## Information

**VIM version**
IM - Vi IMproved 8.1 (2018 May 18, compiled Jul 17 2019 17:46:26)
Included patches: 1-1602

Operating System: rpm based special OS build

## What went wrong

cpp check regular expression is not catching all strings
as the regex is:
```
let l:pattern = '\v^(\f+):(\d+):(\d+): (\w+): (.*) \[(\w+)\]\'
```
while some strings come with a '-' in the last part in the square brackets
for example:
```
/tmp/vaH7BfA/5/main.c:286:8: warning: Local static object: xPrinted [threadsafety-threadsafety]\
```

the fix would be to replace the w with a . (dot)
```
let l:pattern = '\v^(\f+):(\d+):(\d+): (\w+): (.*)\[(.+)\]\'
```

Something went wrong in specifically this place, and I also searched through both open and closed issues for the same problem before reporting a bug here.

Are you having trouble configuring ALE? Try asking for help on [Stack Exchange](https://vi.stackexchange.com/) or perhaps on [Reddit](https://www.reddit.com/r/vim/) instead. The GitHub issue tracker should be used for reporting bugs or asking for new features.

## Reproducing the bug
1. Enable some addons for cpp check for example --addon=threadsafety
to the variable g:ale_c_cppcheck_options

2. Some errors and warnings cannot be shown in the quick fix window and on the side bar
